### PR TITLE
Add ha roles

### PR DESCRIPTION
**Pull request by LongLiveCHIEF**
This pull request adds the `roles` configuration item for controlling scaled instances of GitLab Omnibus.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,6 +2,7 @@
 sudo: false
 dist: trusty
 language: ruby
+
```
Comment (eputnam) [inline .travis.yml]: :(

File: manifests/config.pp [hunk 2]
```diff
@@ -75,6 +75,7 @@
   $backup_cron_enable = $::gitlab::backup_cron_enable
   $backup_cron_minute = $::gitlab::backup_cron_minute
   $backup_cron_hour = $::gitlab::backup_cron_hour
+  $roles = $::gitlab::roles
```
Comment (eputnam) [inline manifests/config.pp]: is there a way to make this var name more descriptive? perhaps `$ha_roles` or something like that?

File: manifests/init.pp [hunk 3]
```diff
@@ -247,6 +247,11 @@
 #   Default: false
 #   Replicate the registry Nginx config from the Gitlab Nginx config.
 #
+# [*roles*]
```
Comment (eputnam) [inline manifests/init.pp]: docs! \o/

File: manifests/config.pp (hunk 2 above)
Comment (LongLiveCHIEF) [reply manifests/config.pp]: For the most part, this module uses variables that are consistent with the names of the config keys expected by the product. This allows users to reference the product documentation for how a config is expected to change the behavior of the product, and removes ambiguity/uncertainty by the implementor on whether or not _our_ name for a config is really _some other_ config for the [gitlab.rb.template](https://gitlab.com/gitlab-org/omnibus-gitlab/blob/master/files/gitlab-config-template/gitlab.rb.template), which is what this module configures.

File: manifests/init.pp (hunk 3 above)
Comment (LongLiveCHIEF) [reply manifests/init.pp]: see my comment above. The standard for this module is to document something that is _not_ a feature of the omnibus package configuration. Documenting every config would be redundant, and could cause unexpected problems when gitlab changes the behavior or possible sub-key values.

There fore, the documentation provided for this param is adequate and in keeping with the standards of this module.

**Comment by LongLiveCHIEF** [comment]
@eputnam are you familiar with the product itself? It seems like some of the review items you raised are indicating that you're unaware that this module is basically puppetizing a chef configuration, and that we're merely automating the generation of that config file according to the docs maintained by the vendor.

We _can't_ be the source of documentation for these configs, because it creates too many issues when debugging configuration problems. When we automate some manual part of the omnibus package/service, _then_ we document it... otherwise we merely indicate an expected type for each param, and in the param docs we point to the usage of that config by the vendor.

It's confusing unless you're familiar with how GitLab Omnibus is administered.

File: .travis.yml (hunk 1 above)
Comment (LongLiveCHIEF) [reply .travis.yml]: Will fix.

**Comment by LongLiveCHIEF** [comment]
@eputnam a good example of how we document non `gitlab.rb.template` features is actually present in one of my other open Merge Requests: [Db indexing for git authorized keys](https://github.com/voxpupuli/puppet-gitlab/pull/177/files).

This is a feature of GitLab that requires manual configuration by the installer, (whereas the majority are handled by the bundled `chef` process according to the settings realized in `gitlab.rb`). As such, I added a section to the `README.md` about enabling it in the module, in addition to  the param docs.

Make sense? Sound good? Maybe we should actually document this distinction in the README so that module consumers don't have to figure out implicitly how our documentation works?

**Comment by LongLiveCHIEF** [comment]
Looks like I need to remove the params and template keys for:

```
redis_master_role
redis_slave_role
redis_sentinel_role
```
Those are no longer viable configs.

**Pull request merged**